### PR TITLE
Remove backport tag from changelog entries

### LINKED ISSUE #141: gen_release_notes backport tag
Backport PRs have branch information in their title (e.g. **[Release 1.23] some issue**). That information should be removed from the release notes.

### PR DESCRIPTION
**Pull request by brooksn**
Fixes #141

### CODE REVIEW

File: repository/repository.go [hunk 1]
```diff
@@ -25,6 +26,8 @@ var repoToOrg = map[string]string{
 	"k3s":  "k3s-io",
 }
 
+var backportRegExp = regexp.MustCompile(`^\s*\[[Rr]elease-?\s*[\w\d\.]*\]\s*(.*)$`)
```
Comment (briandowns) [inline repository/repository.go:29]: What all is this doing?

Comment (brooksn) [reply repository/repository.go:29]: Added a clarifying helper func and tests ea2ccc7

File: repository/repository.go [hunk 2]
```diff
@@ -25,6 +25,15 @@ var repoToOrg = map[string]string{
 	"k3s":  "k3s-io",
 }
 
+// stripBackportTag returns a string with a prefix backport tag removed
+func stripBackportTag(s string) string {
+	if strings.Contains(s, "elease") {
+		s = strings.Split(s, "]")[1]
+	}
+	s = strings.Trim(s, " ")
+	return s
```
Comment (briandowns) [inline repository/repository.go:34]: Looks like we're missing the Title call to make sure the first character is capitalized.

Comment (brooksn) [reply repository/repository.go:34]: strings.Title capitalizes every word, not just the first character. I added a capitalization helper that can be used inside templates in #153

Comment (briandowns) [reply repository/repository.go:34]: I hadn't noticed that behavior before. Cool. Thanks!\

**Review by briandowns — APPROVED** [review]

**Review by nikolaishields — APPROVED** [review]

**Pull request merged**